### PR TITLE
fix java compiler check error

### PR DESCRIPTION
**Pull request by mozhuanzuojing**
1. fix java compiler check error
2. upgrade tycho to 2.6.0

### CODE REVIEW

**Comment by mozhuanzuojing** [comment]
please use `org.eclipse.jdt.core.prefs` for check!
```json
"java.settings.url": "file:///C:/org.eclipse.jdt.core.prefs",
```
[org.eclipse.jdt.core.zip](https://github.com/microsoft/java-debug/files/8517544/org.eclipse.jdt.core.zip)

**Comment by mozhuanzuojing** [comment]
@testforstephen please review this commit

**Comment by testforstephen** [comment]
@mozhuanzuojing Could you pls share more details about the issue you want to fix in this PR?

File: .vscode/launch.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
       "port": 1044,
       "timeout": 30000,
       "sourcePaths": [
-        "${workspaceRoot}/com.microsoft.java.debug.core/main/java",
-        "${workspaceRoot}/com.microsoft.java.debug.plugin/src/main/java"
```
Comment (mozhuanzuojing) [inline .vscode/launch.json:16]: The `workspaceRoot` parameter is deprecated in vscode. `workspaceFolder` is now recommended

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java [hunk 2]
```diff
@@ -85,18 +85,18 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
         ISourceLookUpProvider sourceProvider = context.getProvider(ISourceLookUpProvider.class);
         sourceProvider.initialize(context, options);
         // If the debugger and debuggee run at the different JVM platforms, show a warning message.
-        if (debugSession != null) {
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java:88]: compiler check: `debugSession` can not null here. Don't judge null.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java [hunk 3]
```diff
@@ -93,7 +93,7 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
                     Value sizeValue = null;
                     if (value instanceof ArrayReference) {
                         indexedVariables = ((ArrayReference) value).length();
-                    } else if (value instanceof ObjectReference && DebugSettings.getCurrent().showLogicalStructure && engine != null) {
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java:96]: compiler check: `engine` can not null here. Don't judge null.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java [hunk 4]
```diff
@@ -147,7 +147,7 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
                     } else {
                         String typeString = "";
                         try {
-                            typeString = variableFormatter.typeToString(value == null ? null : value.type(), options);
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java:150]: compiler check: `value` can not null here. Don't judge null.

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/InlineValueHandler.java [hunk 5]
```diff
@@ -166,7 +166,7 @@ private static IMethod findMethodInLocalTypes(IMethod enclosingMethod, int stopp
      */
     private static Position getPosition(IBuffer buffer, int offset) {
         int[] result = JsonRpcHelpers.toLine(buffer, offset);
-        if (result == null && result.length < 1) {
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/InlineValueHandler.java:169]: compiler check:  if `result` is null,then `result.lenth` will throw NullPointerException

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java [hunk 6]
```diff
@@ -276,7 +276,7 @@ private static IClassFile resolveClassFile(String uriString) {
         }
         try {
             URI uri = new URI(uriString);
-            if (uri != null && JDT_SCHEME.equals(uri.getScheme()) && "contents".equals(uri.getAuthority())) {
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java:279]: compiler check: `uri` can not null here. Don't judge null.

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/VariablesRequestHandler.java [hunk 7]
```diff
@@ -259,7 +259,7 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
             String evaluateName = null;
             if (javaVariable.evaluateName == null || (containerEvaluateName == null && containerNode.getProxiedVariable() instanceof ObjectReference)) {
                 // Disable evaluate on the method return value.
-                evaluateName = null;
```
Comment (mozhuanzuojing) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/VariablesRequestHandler.java:262]: compiler check: `evaluateName` always is null.

File: pom.xml [hunk 8]
```diff
@@ -11,7 +11,7 @@
     <properties>
         <base.name>Java Debug Server for Visual Studio Code</base.name>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <tycho-version>1.5.1</tycho-version>
+        <tycho-version>2.6.0</tycho-version>
```
Comment (mozhuanzuojing) [inline pom.xml:14]: Interested can experience mixed-reactor-build-support
https://github.com/eclipse/tycho/blob/tycho-2.7.x/RELEASE_NOTES.md#mixed-reactor-build-support

**Pull request closed without merging**